src/opm/output/eclipse/WriteInit.cpp

```diff
@@ -390,7 +390,17 @@ namespace {
 
             if (!fp.has_double(prop.name))
                 continue;
+            const auto& unit_system = fp.getUnitSystem();
             auto data = fp.get_double(prop.name);
+            const auto& kw_info = ::Opm::Fieldprops::keywords::global_kw_info<double>(prop.name);
+            if(kw_info.unit){
+                const auto& dim = unit_system.parse( *kw_info.unit );
+                for(auto& val: data){
+                    val = dim.convertSiToRaw(val);
+                }
+            }else{
+                // option kw_info.unit not set assume unit conversion
+            }
```

Comment:
Since you also remove the unit conversion from the `needDflt` branch in `writeDoubleCellProperties()` you really need this kind of treatment in `writeCellDoublePropertiesWithDefaultFlag()` too.  Perhaps we could add a new helper function of the form
```C++
std::vector<double>
getUnitConvertedPropertyValue(const std::string& propertyName,
                              const ::Opm::FieldPropsManager& fp)
{
    auto data = fp.get_double(propertyName);
    const auto kw_info = ::Opm::Fieldprops::keywords::global_kw_info<double>(propertyName);

    if (kw_info.unit.has_value()) {
        const auto dim = fp.getUnitSystem().parse(*kw_info.unit);
        for (auto& val : data) {
            val = dim.convertSiToRaw(val);
        }
    }

    return data;
}
```
and then call that helper function from both branches in `writeDoubleCellProperties()`, i.e., instead of
```C++
auto data = fp.get_double(prop.name);
```
we write
```C++
auto data = getUnitConvertedPropertyValue(prop.name, fp);
```

Reply:
Changed to you version.